### ISSUE TITLE
[EKS] [request]: Move VPC CNI configuration settings to ConfigMap

### ISSUE THREAD

**Issue by mikestef9** [aws/containers-roadmap#865]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
Add an option for the VPC CNI to pull configuration settings from a Kubernetes ConfigMap

**Which service(s) is this request for?**
EKS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
Today, all VPC CNI configuration settings must be specified as environment variables on the aws-node daemonset. This can make upgrading difficult, as a new version of the VPC CNI may have a different manifest, but has all default settings. We should be able to follow upgrade instructions without worrying about configuration settings.

**Are you currently working around this issue?**
Download new manifest file, manually edit environment variables, then apply to cluster.

**Comment by muvaf** [comment]
@janwillies does that mean if we pushed just a `ConfigMap`, that'd be all users need to do to get CNI?

**Comment by sftim** [comment]
Hi. Are there any details about what the format of the ConfigMap might look like?

**Comment by TBBle** [comment]
https://github.com/aws/amazon-vpc-cni-k8s/pull/1088 was an approved but not merged (not clear why) PR, which introduces a ConfigMap like this: https://github.com/aws/amazon-vpc-cni-k8s/pull/1088/files#diff-94446c74858d682e37df1247778b6245243289fb4db689921a0bfa298cd47bc5

There wasn't any objection raised to the format that I can see, so that seems like the most-likely structure to go ahead, but the PR needs to be updated to the current state of that project first.

**Comment by sftim** [comment]
https://github.com/aws/amazon-vpc-cni-k8s/pull/1088 doesn't define a ConfigMap for the CNI plugin to load; rather, it uses Kubernetes APIs to separate the storage of those environment variables from the Pod template in the sample DaemonSet. Maybe that's good enough for what @mikestef9 had in mind?

**Comment by TBBle** [comment]
Yeah, @mikestef9 agreed that that PR resolved his need: https://github.com/aws/amazon-vpc-cni-k8s/pull/1088#discussion_r503044669
